test/e2e/commands/ingest_xmlupload/test_ingest_xmlupload.py

```diff
@@ -0,0 +1,36 @@
+from pathlib import Path
+from typing import Iterator
+
+import pytest
+
+from dsp_tools.cli.args import ServerCredentials
+from dsp_tools.commands.ingest_xmlupload.upload_files.upload_files import upload_files
+from dsp_tools.commands.project.create.project_create import create_project
+from test.e2e.setup_testcontainers import get_containers
+
+CREDS = ServerCredentials("root@example.com", "test", "http://0.0.0.0:3333")
+XML_FILE = Path("testdata/xml-data/test-data-e2e.xml")
+
+
+@pytest.fixture()
+def _create_project() -> Iterator[None]:
+    with get_containers():
+        success = create_project(Path("testdata/json-project/test-project-e2e.json"), CREDS, verbose=True)
+        assert success
+        yield
+
+
+@pytest.mark.usefixtures("_create_project")
+def test_ingest_upload(caplog: pytest.LogCaptureFixture) -> None:
+    _test_upload_step(caplog)
+
+
+def _test_upload_step(caplog: pytest.LogCaptureFixture) -> None:
+    success = upload_files(XML_FILE, CREDS)
+    assert success
+    log_messages = [rec.message for rec in caplog.records]
+    assert log_messages[0] == f"Found 2 files to upload onto server {CREDS.dsp_ingest_url}."
+    assert log_messages[-3] == "Uploaded file 'testdata/bitstreams/test.jpg'"
+    assert log_messages[-2] == "Uploaded file 'testdata/bitstreams/test.pdf'"
+    assert log_messages[-1] == f"Uploaded all 2 files onto server {CREDS.dsp_ingest_url}."
+    caplog.clear()
```

Comment:
We discussed that it would be better to check on the filesystem if the files have arrived in the temporary folder. I like that idea, but then we always have to run our e2e tests within the test containers. Because the path to the temp folder is defined in the setup of our testcontainers: https://github.com/dasch-swiss/dsp-tools/blob/d1a01fc1e5fcc2b337f452ff97535a1513a35d41/test/e2e/setup_testcontainers.py#L15-L18

It won't be possible any more to run our e2e tests against a stack that was manually started, e.g. from a clone of the DSP-API repository. 

I think it's worth to pay that price. Because our e2e tests are anyways set up to rely on the testcontainers. What do you think, @seakayone ?

Reply:
You can build the API docker image locally (similar to starting the API locally) and then specify in the test setup to use that version of the API image. This is admittedly not quite as simple as just running the API and then the tests - but it should still be easy enough.

Reply:
done with https://github.com/dasch-swiss/dsp-tools/commit/7c73b9167af073242c5bcea6f11ff9aabb979228